### ISSUE TITLE
Cant weights to change in BatchSemiHardTripletLoss/Training does not alter weights

### ISSUE THREAD

**Issue by zeno17** [UKPLab/sentence-transformers#448]
Hi,

Im trying to alter the embeddings of SBert based on some data. Ideally, i would have 2 sentences and a similarity score to finetune on but i dont and now im trying to use a proxy which is a sentence and a label (0 to 10). I dont actually care about it getting the labels correct, but given that it is just a proxy for some other process that i am trying to simulate (and of which data is hard to get).

This is an example of the data that im trying to read in (shown in a dataframe)
```
  | Sentence | Score
-- | -- | --
Some test sentence | 0.0
Just another sentence to see if it works | 1.0
Well, maybe this one will actually work | 6.0
```
Then i have the following setting
base_model = SentenceTransformer('distiluse-base-multilingual-cased')
model_mli = SentenceTransformer('distiluse-base-multilingual-cased')
qs_reader = LabelSentenceReader(finetunedir, 1, 0)
train_data = SentenceLabelDataset(qs_reader.get_examples('data.tsv'), model_mli)
train_dataloader = DataLoader(train_data, shuffle=True)
train_loss = BatchSemiHardTripletLoss(model_mli, margin=0.01)

Then i try to train it using the following settings (ive used more hyperparameter settings, but ive been trying absolute extreme ones that in practice should wreck any prior weights just to see if it does anything) :

```
model_mli.fit(train_objectives=[(train_dataloader, train_loss)],
                                  epochs=1,
                                  scheduler= 'constantlr',
                                  evaluation_steps=1000,
                                  evaluator = None, 
                                  optimizer_params = {"lr": 1},
                                  output_path=finetunedir + 'test')
```
After training i try to obtain an encoding from both the old model and the one that should be finetuned:
`
enc1 = base_model.encode(['Dit is een testzin die helemaal nergens over gaat. Hoe cool is dat?'])
enc2 = model_mli.encode(['Dit is een testzin die helemaal nergens over gaat. Hoe cool is dat?'])
`

followed by a sanity check 

`np.sum(enc1 == enc2)`

If this sanity check is 512 it means that weights did not move at all.

Unfortunately, trying many hyperparameters and increasing the dataset upto ~6000 sentence-label pairs does not seem to be able to change anything. So im wondering if theres anything wrong with my setup
